app/scripts/components/common/banner/index.tsx

```diff
@@ -1,78 +1,136 @@
 import React, { useState } from 'react';
-import { Icon } from '@trussworks/react-uswds';
+import { decode } from 'he';
 import {
   USWDSBanner,
-  USWDSBannerContent
+  USWDSBannerContent,
+  USWDSBannerButton,
+  USWDSBannerFlag,
+  USWDSBannerHeader,
+  USWDSBannerIcon,
+  USWDSBannerGuidance,
+  USWDSMediaBlockBody
 } from '$components/common/uswds/banner';
 
-const BANNER_KEY = 'dismissedBannerUrl';
-
-function hasExpired(expiryDatetime) {
-  const expiryDate = new Date(expiryDatetime);
-  const currentDate = new Date();
-  return !!(currentDate > expiryDate);
+interface Guidance {
+  left?: GuidanceContent;
+  right?: GuidanceContent;
 }
 
-enum BannerType {
-  info = 'info',
-  warning = 'warning'
+interface GuidanceContent {
+  icon?: string;
+  iconAlt?: string;
+  title?: string;
+  text?: string;
 }
 
-const infoTypeFlag = BannerType.info;
 interface BannerProps {
-  appTitle: string;
-  expires: Date;
-  url: string;
-  text: string;
-  type?: BannerType;
+  headerText?: string;
+  headerActionText?: string;
+  ariaLabel?: string;
+  flagImgAlt?: string;
+  leftGuidance?: GuidanceContent;
+  rightGuidance?: GuidanceContent;
+  className?: string;
+  defaultIsOpen?: boolean;
+  contentId?: string;
 }
 
+const DEFAULT_HEADER_TEXT =
+  'An official website of the United States government';
+
+const DEFAULT_HEADER_ACTION_TEXT = "Here's how you know";
+
+const DEFAULT_GUIDANCE: Guidance = {
+  left: {
+    title: 'Official websites use .gov',
+    text: 'A .gov website belongs to an official government organization in the United States.',
+    iconAlt: 'Dot gov icon',
+    icon: '/img/icon-dot-gov.svg'
+  },
+  right: {
+    title: 'Secure .gov websites use HTTPS',
+    text: `
+        A <strong>lock</strong> or <strong>https://</strong> means you've safely
+        connected to the .gov website. Share sensitive information only on
+        official, secure websites.
+      `,
+    iconAlt: 'HTTPS icon',
+    icon: '/img/icon-https.svg'
+  }
+};
+
+const GuidanceBlock = ({
+  content,
+  className
+}: {
+  content: GuidanceContent;
+  className?: string;
+}) => (
+  <USWDSBannerGuidance className={className}>
+    <USWDSBannerIcon src={content.icon} alt={content.iconAlt ?? ''} />
+    <USWDSMediaBlockBody>
+      <p>
+        <strong>{content.title}</strong>
+        <br />
+        <span
+          dangerouslySetInnerHTML={{ __html: decode(content.text ?? '') }}
```

Comment:
Thanks @hanbyul-here! Does this mean we will restrict the Banner component to only allow users to specify a predefined set of content types (e.g., a url prop and a text prop) for which we would then render the appropriate html tags (`<p />`, `<a />`) directly within the Banner?

Reply:
No - I meant that user can pass the contents as markdown like `::markdown ### heading` - I see that you used some html class for mock data https://github.com/NASA-IMPACT/veda-ui/blob/main/mock/veda.config.js#L133 I think since markdown accepts html as a part of markdown, it should be handled ok with markdown parser? I mainly want all the user-configured inputs for our component contents to be in a coherent format, and the parsing of the contents to be handled in the same place.

Reply:
Created a ticket for myself: https://github.com/NASA-IMPACT/veda-ui/issues/1348